modules/mobile/spec/request/v1/messages_request_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../../support/iam_session_helper'
+require_relative '../../support/mobile_sm_client_helper'
+
+RSpec.describe 'Mobile Messages Integration', type: :request do
+  include Mobile::MessagingClientHelper
+  include SchemaMatchers
+
+  let(:user_id) { '10616687' }
+  let(:inbox_id) { 0 }
+  let(:message_id) { 573_059 }
+  let(:va_patient) { true }
+
+  before do
+    allow_any_instance_of(MHVAccountTypeService).to receive(:mhv_account_type).and_return(mhv_account_type)
+    allow(Mobile::V0::Messaging::Client).to receive(:new).and_return(authenticated_client)
+    iam_sign_in(build(:iam_user, iam_mhv_id: '123'))
+  end
+
+  context 'Premium User' do
```

Comment:
I see that the old specs also mention basic and advanced users and they don't have access. Is that also true of the new api? Are they denied access by the parent controller (MessagingController)?

Reply:
I believe so. From what I can tell, they're still checking for the same `mhv_messaging` access, which looks like it's a premium service